scripts/set_maven_version.sh

```diff
@@ -20,7 +20,12 @@ POM_FILES=(
   examples/pom.xml
 )
 
-for name in ${POM_FILES[*]};
-do
-  mvn versions:set -D generateBackupPoms=false -D newVersion=$VERSION -f $name
-done
+node -e "$(cat <<EOF
+  const parts = process.argv[1].split('.').map(part => parseInt(part, 10));
+  parts[1]--;
+  const previousMajorVersion = parts.join('.');
```

Comment:
This is not gonna work, on the `main` branch the version is `1.X.Y-SNAPSHOT`, we need to drop the -snapshot part there.